includes/data-stores/class-wc-admin-reports-orders-data-store.php

```diff
@@ -70,24 +70,28 @@ public function __construct() {
 	protected function get_sql_query_params( $query_args ) {
 		global $wpdb;
 		$order_stats_lookup_table = $wpdb->prefix . self::TABLE_NAME;
+		$operator                 = $this->get_match_operator( $query_args );
+		$where_subquery           = array();
 
 		$sql_query_params = $this->get_time_period_sql_params( $query_args, $order_stats_lookup_table );
 		$sql_query_params = array_merge( $sql_query_params, $this->get_limit_sql_params( $query_args ) );
 		$sql_query_params = array_merge( $sql_query_params, $this->get_order_by_sql_params( $query_args ) );
 
 		$status_subquery = $this->get_status_subquery( $query_args );
 		if ( $status_subquery ) {
-			$sql_query_params['where_clause'] .= " AND {$status_subquery}";
+			$where_subquery[] = $status_subquery;
```

Comment:
I think we'll need to apply this status subquery without an `OR` match operator unless the status query is explicitly set.  Since `get_status_subquery()` by default returns the 'normal' statuses, an `OR` operator would also return excluded statuses even if the user didn't add a status filter.

The code in the order stats data store probably does a better job of explaining this than I did - https://github.com/woocommerce/woocommerce-admin/blob/bc5dc57703b856e95c512883ab85ffa0e4639f41/includes/data-stores/class-wc-admin-reports-orders-stats-data-store.php#L173

Reply:
@joshuatf Thanks. Restored the AND for status in e192317.

Reply:
@joshuatf I've updated the PR with that logic. However, what I noticed is that the default query excludes trash, pending, failed and cancelled orders. When the OR is added for an Any + any status then the default excluded statuses are not excluded. 

As a user I would expect the original exclusion to persist unless I had explicitly added them. To provide that behaviour we would need to mask the default excludes to remove explicitly requested statuses then include both subqueries. Is that the expected/desired behaviour?

Reply:
I see what you mean here.  I think that could be expected behavior, but I can definitely see the argument for wanting to continue to exclude statuses.  For anyone else following here, if you applied the following filter:

<img width="455" alt="Screen Shot 2019-06-17 at 12 56 14 PM" src="https://user-images.githubusercontent.com/10561050/59579427-5f36d500-90ff-11e9-8f7e-e2c47de3cd25.png">

This could potentially allow `failed` (or any of your other excluded statuses) to now show if they had a returning customer order with a `failed` status.

If we did opt to change this, I think the best method would be to:
1. Continue appending the current and/or status subquery.
2. Check if the applied status filter is in the array of excluded statuses (i.e., they're explicitly querying excluded statuses).
3. If not- append an `AND` query to negate any excluded statuses.
4. If yes- do not append further queries.

Reply:
@joshuatf Thanks for the feedback. That logic makes sense. If we implement it we end up with a situation where a user adds the following with `Any`

- Status: completed
- Product: Product X

The rows of completed Product X are already included in the completed status selection and the product filter has no effect. Either way we are going to end up with unexpected results for some users.

I'll merge this and if user feedback warrants we can change how the filters interact at a later date.